### ISSUE TITLE
Update CanOpenNode to latest version

### ISSUE THREAD

**Issue by MaJerle** [CANopenNode/CanOpenSTM32#35]
@HamedJafarzadeh  Are you ready to do it? :)

**Comment by HamedJafarzadeh** [comment]
Well, I'm just back from holidays, give me some time and I'll do it :) But it seems there are no breaking changes so It should work right away. I'll confirm that though.

**Comment by SWolfSchunk** [comment]
Well as of now the Submodule is still not updated and nearly 5 years behind.

**Comment by MaJerle** [comment]
I agree, @HamedJafarzadeh could you please check this part? Maybe we start new branch first, merge latest submodule to verify compilation, then merge to main branch?

I think many problems reported are due to the old version.

@SWolfSchunk if you have a suggestion or would like to try, let me know. I'm open to start with clean branch

**Comment by SWolfSchunk** [comment]
I can verify that it builds with the latest master.
So for a new branch I think this should be ok. 
Correct operation is another topic. But I think this should be tested in the new branch.

**Comment by CANopenNode** [comment]
Yes please, update to the latest CANopenNode.

**Comment by MaJerle** [comment]
https://github.com/CANopenNode/CanOpenSTM32/commit/d915161e2f4457b0ba736408a6b8e46496c66e6e
